src/mbgl/gfx/upload_pass.hpp

```diff
@@ -124,7 +147,7 @@ class UploadPass {
                                           TexturePixelType,
                                           TextureChannelDataType) = 0;
 
-    virtual std::shared_ptr<gfx::Texture2D> createTexture2D(const PremultipliedImage& image) = 0;
+    virtual gfx::Texture2DPtr createTexture2D() = 0;
```

Comment:
How about moving this method to Context? the merge with Raster will be easier

Reply:
on it...

Reply:
Not sure that will be better, I'm having to add a `Context` parameter to a bunch of `upload` methods.

Reply:
I suppose those methods have uploadPass? Maybe create a getter for Context in the UploadPass?

Reply:
oh, I missed that I can get it from the encoder in there, that should be better